API_Auto/src/test/java/APIAssignments/Items.java

```diff
@@ -0,0 +1,97 @@
+package APIAssignments;
+
+import io.restassured.RestAssured;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+import org.apache.log4j.PropertyConfigurator;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static io.restassured.RestAssured.*;
+
+public class Items {
+
+    private static String url;
+    private String productId;
+    private String productName;
+    private String productPrice;
+    private String productBrand;
+
+    public Items() {
+        url = ReusableMethod.getUrl();
+    }
+
+    public Items(String productId, String productName, String productPrice, String productBrand) {
+        this.productId = productId;
+        this.productName = productName;
+        this.productPrice = productPrice;
+        this.productBrand = productBrand;
+    }
+
+    private static final Logger log = Logger.getLogger("Product.class");
+
+    @BeforeTest
+    public void getLoggerDisplay() {
+        PropertyConfigurator.configure("log4j2.properties");
+    }
+
+    @Test(priority = 1)
+    public void checkStatusCode() {
+        {
+            baseURI = ReusableMethod.getUrl();
+            Response response = RestAssured.get(ReusableMethod.getUrl()).then().extract().response();
+            Assert.assertEquals(response.getStatusCode(), 200);
+            log.info("Status code is " + response.getStatusCode());
+        }
+    }
+
+    @Test(priority = 2)
+    public void checkProductList() {
+        var getList = given().when().get(url).then().log().all().toString();
+        log.info("Product List is " + getList);
+    }
+
+    @Test(priority = 3)
+    public void checkContent() {
+        Response response = RestAssured.get(url).then().extract().response();
+        JsonPath jsonObject = new JsonPath(response.asString());
+        List<Items> productList = jsonObject.getList("products");
+        List<Items> mockData = new ArrayList<>();
+        Items productLists = new Items("1", "Blue Top", "Rs. 500", "Polo");
+        mockData.add(productLists);
+        productLists = new Items("2", "Men Tshirt", "Rs. 400", "H&M");
```

Comment:
Use getter and setter to get and set the values

Reply:
done